src/components/Card.jsx

```diff
@@ -0,0 +1,136 @@
+import Arrow from "../assets/icon-arrow.svg";
+import { useEffect, useState } from "react";
+
+import FormInput from "./FormInput";
+import AgeResults from "./AgeResults";
+const monthsUtil = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+const Card = () => {
+  const [day, setDay] = useState("");
+  const [month, setMonth] = useState("");
+  const [year, setYear] = useState("");
+  const [result, setResult] = useState({
+    year: "-",
+    month: "-",
+    days: "-",
+  });
+  const [error, setError] = useState({
+    error: false,
+    msj: "",
+  });
+
+  useEffect(() => {
+    if (day.length == 0 || month.length == 0 || year.length == 0) {
+      setError({
+        error: true,
+        msj: "This field is required",
+      });
+    } else if (
+      day <= 0 ||
+      day > 31 ||
+      month <= 0 ||
+      month > 12 ||
+      year <= 0 ||
+      year > 2023
+    ) {
+      setError({
+        error: true,
+        mensaje: "Must be a valid date",
+      });
+    } else {
+      setError({
+        error: false,
+        msj: "",
+      });
+    }
+  }, [day, month, year]);
+  const ageCalculator = () => {
+    if (error.error) {
+      return;
+    }
+    const currentDate = new Date();
+    const currentYear = currentDate.getFullYear();
+    const currentMonth = currentDate.getMonth() + 1;
+    const currentDay = currentDate.getDate();
+
+    let resultAge = currentYear - Number(year);
+    let resultMonth;
+    let resultDay;
+    if (currentMonth >= month) {
+      resultMonth = currentMonth - Number(month);
+    } else {
+      resultAge--;
+      resultMonth = 12 + currentMonth - Number(month);
+    }
+
+    if (currentDay >= day) {
+      resultDay = currentDay - Number(day);
+    } else {
+      resultMonth--;
+      let days = monthsUtil[currentMonth - 2];
```

Comment:
There's probably an easier way to handle this; pseudocode: `const differenceInDays = dateOne - dateTwo;`